pkg/piecestore/rpc/server/psdb/psdb.go

```diff
@@ -247,3 +274,75 @@ func (db *DB) DeleteTTLByID(id string) error {
 	}
 	return err
 }
+
+// AddMIB adds MIB into database by date
+func (db *DB) AddBwUsageTbl(size, unixtimenow int64) (err error) {
+	defer db.locked()()
+	t := time.Now()
+	fmt.Println("time now =", t.Unix())
+
+	daystartunixtime := time.Date(t.Year(), t.Month(), t.Day(), 0, 0, 0, 0, t.Location()).Unix()
+	fmt.Println("daystarttime =", daystartunixtime)
+
+	dayendunixtime := time.Date(t.Year(), t.Month(), t.Day(), 24, 0, 0, 0, t.Location()).Unix()
+	fmt.Println("dayendtime =", dayendunixtime)
+
+	var getSize int64
+	if (unixtimenow >= daystartunixtime) && (unixtimenow <= dayendunixtime) {
+		err = db.DB.QueryRow(`SELECT size FROM bwusagetbl WHERE daystartdate <= ? AND ? <= dayenddate`, unixtimenow, unixtimenow).Scan(&getSize)
+		log.Println("KISHORE --> getSize + size = ", getSize, size, (getSize + size))
+		switch {
+		case err == sql.ErrNoRows:
+			fmt.Println("New day starting new entry ", err)
+			zap.S().Warn("New day starting new entry %+v", err)
+			_, err = db.DB.Exec("INSERT INTO bwusagetbl (size, daystartdate, dayenddate) VALUES (?, ?, ?)", size, daystartunixtime, dayendunixtime)
+			return err
+		case err != nil:
+			fmt.Println("Invalid query return", err)
+			zap.S().Errorf("Invalid query return %v", err)
+			return err
+		default:
+			getSize = size + getSize
+			_, err = db.DB.Exec("UPDATE bwusagetbl SET size = ? WHERE daystartdate = ?", getSize, daystartunixtime)
+			zap.S().Info("Successfully written the into the bwusagetbl size = ", getSize)
+			log.Println("KISHORE --> Successfully written size = ", getSize)
+			return err
+		}
+	}
+	fmt.Println("Invalid time passed", unixtimenow)
+	zap.S().Errorf("Invalid time passed %v", unixtimenow)
+	return err
+}
+
+// GetMIBByDate finds the so far bw used by date and return it
+func (db *DB) GetBwUsageTbl(t time.Time) (size int64, err error) {
```

Comment:
`func (db *DB) TotalBandwidthUsedSince(t time.Time) (int64, error) {`, return in bytes

Reply:
I am really not so good with namings :-). I like the what you suggested. Changed as below: 

// _GetMIBByDate_ finds the so far bw used by date and return it
func (db *DB) _GetBwUsageTbl_(t time.Time) (size int64, err error) --> 
**changed to** 
// **TotalBandwidthUsedOnADay** finds the so far bw used by day and return it
func (db *DB) **TotalBandwidthUsedOnADay**(t time.Time) (size int64, err error) {

/ _BandwidthUsage_ each row in the bwusagetbl contains the total bw used perday
func (db *DB) _BandwidthUsage_(startdate time.Time, enddate time.Time) (totalbwusage int64, err error) { -->
**Changed to** 
// **TotalBandwidthUsedSince** each row in the bwusagetbl contains the total bw used perday
func (db *DB) **TotalBandwidthUsedSince**(startdate time.Time, enddate time.Time) (totalbwusage int64, err error) {

Reply:
maybe:
`DayBandwidth(day time.Time)`
`BandwidthBetween(startday, endday time.Time)`